### DISCUSSION TITLE
Message datas are not shown in Wireshark?

### DISCUSSION THREAD

**Discussion by mirusu400** [eProsima/Fast-DDS#4252] [Q&A]
Hello.

I'm trying to running a simple C++ example by [this document](https://fast-dds.docs.eprosima.com/en/latest/fastdds/getting_started/simple_app/simple_app.html#cmakelists-txt), and trying to capture packets between publisher and subscriber

https://github.com/eProsima/Fast-DDS/assets/25397908/54c9e5bd-b619-47c7-9b7b-6b5b2491a21d

[question_packet.zip](https://github.com/eProsima/Fast-DDS/files/13956347/question_packet.zip)
^ Here are some packets I've captured.

I can capture `INFO_TS`, `DATA(p)`, `INFO_DST`, `HEARTBEST`, `ACKNACK` packets, but I can't find any packets about serving real messages (Such as `CustomHello` - In this case).

I've searching some documents about how the messages are served, but I can't find any informations about how publisher/subscriber managing their message data packets.

Is message datas are really serving via `rtps`? if not, How they serve their datas? Also, How can I capture these packets using `wireshark`?

**Comment by JesusPoderoso** [comment]
Hi @mirusu400, thanks for using Fast DDS.

User data transmissions are represented in Wireshark as `DATA(w)` and `DATA(r)`.

Fast DDS [uses by default](https://fast-dds.docs.eprosima.com/en/latest/fastdds/rtps_layer/rtps_layer.html#managing-the-builtin-transports) shared memory prior than UDPv4 as builtin transport.
It may happen that the data transmission is performed through shared memory, and Wireshark is not able to track it in real-time. 
You can configure the transport descriptor's attribute `rtps_dump_file_` and follow [this section indications](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/shared_memory/shared_memory.html?highlight=wireshark#sharedmemtransportdescriptor) to dump the traffic to a file, where you will be able to analyze the user traffic.
On the other hand, you can configure the participant transport to force the communication to be through UDP, so you can capture those packages in-real-time in Wireshark.

Following your suggested guidelines:

> I'm trying to running a simple C++ example by [this document](https://fast-dds.docs.eprosima.com/en/latest/fastdds/getting_started/simple_app/simple_app.html#cmakelists-txt), and trying to capture packets between publisher and subscriber

You should include the following code lines in the `init()` method, between lines `123-124` (before creating the participant). They disable default builtin transports, and set the user transport to use `UDPv4`

```
participantQos.transport().use_builtin_transports = false;
auto udp_transport = std::make_shared<UDPv4TransportDescriptor>();
participantQos.transport().user_transports.push_back(udp_transport);
```

TIP: You can try to apply the following filter in Wireshark to display only `HelloWorld` user data: `rtps.param.topicName == HelloWorldTopic`.
